synthesizer/program/src/traits/stack_and_registers.rs

```diff
@@ -150,6 +151,84 @@ pub trait StackTrait<N: Network> {
     ) -> Result<Record<N, Plaintext<N>>>;
 }
 
+/// Are the two types either the same, or both structurally equivalent `PlaintextType`s?
+pub fn register_types_structurally_equivalent<N: Network>(
```

Comment:
Is it necessary to pass in two stacks or can we just use one and resolve external stacks via `get_external_stack`?
This would have the side-effect of checking imports are defined for any external stack you get.

Ditto for below.

Reply:
Understood, sort of like
A     B
  \    /
    C
    
where A and B are structurally equivalent.
This would be an interesting test case to have.